### PR TITLE
use the bootstrapped compiler for `test-float-parse` test

### PR DESCRIPTION
**Pull request by onur-ozkan**
Fixes https://github.com/rust-lang/rust/pull/122709#issuecomment-2327259336.

Blocker for https://github.com/rust-lang/rust/pull/122709

### CODE REVIEW

File: src/bootstrap/src/core/build_steps/test.rs [hunk 1]
```diff
@@ -3526,12 +3526,10 @@ impl Step for TestFloatParse {
 
     fn run(self, builder: &Builder<'_>) {
         let bootstrap_host = builder.config.build;
-        let compiler = builder.compiler(0, bootstrap_host);
+        let compiler = builder.compiler(builder.top_stage, bootstrap_host);
         let path = self.path.to_str().unwrap();
         let crate_name = self.path.components().last().unwrap().as_os_str().to_str().unwrap();
```
Comment (Kobzol) [inline src/bootstrap/src/core/build_steps/test.rs:3532]: Should this really be removed? I tried to clean `build` and run `./x test src/etc/test-float-parse` from scratch, and got this:
```
error[E0463]: can't find crate for `core`
  |
  = note: the `x86_64-unknown-linux-gnu` target may not be installed
  = help: consider adding the standard library to the sysroot with `x build library --target x86_64-unknown-linux-gnu`
  = help: consider building the standard library from source with `cargo build -Zbuild-std`

error[E0463]: can't find crate for `std`
  |
  = note: the `x86_64-unknown-linux-gnu` target may not be installed
  = help: consider adding the standard library to the sysroot with `x build library --target x86_64-unknown-linux-gnu`
  = help: consider building the standard library from source with `cargo build -Zbuild-std`
```

Comment (onur-ozkan) [reply src/bootstrap/src/core/build_steps/test.rs:3532]: Right, I was tested it only for ci-rustc case. Thanks!

**Pull request merged**